### DISCUSSION TITLE
Weird duplicated clicks

### DISCUSSION THREAD

**Discussion by garethjax** [shlinkio/shlink#1727] [Help wanted]
#### How Shlink is set up

* Shlink Version: 3.0.3
* PHP Version: 8.0
* How do you serve Shlink: Self-hosted apache
* Database engine used: SQLite (x.y.z)

#### Summary

In general it works very well, but there have always been some cases when the same click is tracked twice.

For example this is me:

<img width="825" alt="image" src="https://user-images.githubusercontent.com/3459602/226946344-a0520493-7311-480b-b9e5-41f1b65e9bec.png">

I've just clicked on a link on a web page.

It is pretty random, i've tried to copy the link and open it directly, there was only a single click tracked but of course the referree was "direct".

And when i've tried again to click on the page, the duplication is missing.

<img width="886" alt="image" src="https://user-images.githubusercontent.com/3459602/226947533-40bb44ac-8077-4607-8ada-84aaf2258c60.png">

**Comment by acelaya** [comment]
This is pretty random, so before spending any time investigating, I would strongly advice two things:

- Try updating to the latest version, to discard it's something which is currently fixed.
- Use a proper database engine, as it's discouraged to use SQLite for anything other than trying Shlink to see if it fits your needs, but should never be used in production.

**Comment by garethjax** [comment]
understandable!

I have two issues:
- i've tried to upgrade to 3.5.3, but i'm getting a doctrine error and it cannot upgrade itself. Maybe i should open a different issue about it.
- i've seen some thread and it seems there's no "migration" procedure from sqlite to mariadb: will i lose every tracked data ?
thanks!

**Comment by acelaya** [comment]
> - i've tried to upgrade to 3.5.3, but i'm getting a doctrine error and it cannot upgrade itself. Maybe i should open a different issue about it.

Follow the instructions here https://shlink.io/documentation/update-your-instance/. Pay special attention to this sentence: `If your Shlink instance is out of date, preferably update Shlink in sequential order using the latest patch release for every minor version. For example, to go from version 3.2.1 to version 3.4.0, first update to 3.3.2, and then 3.4.0.`

If that doesn't help, then yes, open another issue.

> - i've seen some thread and it seems there's no "migration" procedure from sqlite to mariadb: will i lose every tracked data ?

The way to do this is through the import tool https://shlink.io/documentation/advanced/import-short-urls/#shlink-instance

This might even allow you to bypass the first problem. Just setup a clean up-to-date instance, and import urls from the old one.
